--- moai-dev-Version-1.5.2/3rdparty/sdl2-2.0.0/src/render/opengl/SDL_render_gl.c.orig	2025-11-16 22:18:33.304947821 +0100
+++ moai-dev-Version-1.5.2/3rdparty/sdl2-2.0.0/src/render/opengl/SDL_render_gl.c	2025-11-16 22:18:46.039406884 +0100
@@ -319,7 +319,7 @@ GL_ResetState(SDL_Renderer *renderer)
 }
 
 static void APIENTRY
-GL_HandleDebugMessage(GLenum source, GLenum type, GLuint id, GLenum severity, GLsizei length, const char *message, void *userParam)
+GL_HandleDebugMessage(GLenum source, GLenum type, GLuint id, GLenum severity, GLsizei length, const GLchar *message, const void *userParam)
 {
     SDL_Renderer *renderer = (SDL_Renderer *) userParam;
     GL_RenderData *data = (GL_RenderData *) renderer->driverdata;
