--- yosys-yosys-0.29/abc/src/misc/util/abc_global.h.orig	2023-05-28 15:09:21.134631999 +0200
+++ yosys-yosys-0.29/abc/src/misc/util/abc_global.h	2023-05-28 15:09:51.882987655 +0200
@@ -94,8 +94,6 @@
 ///                         PARAMETERS                               ///
 ////////////////////////////////////////////////////////////////////////
 
-ABC_NAMESPACE_HEADER_START
-
 ////////////////////////////////////////////////////////////////////////
 ///                         BASIC TYPES                              ///
 ////////////////////////////////////////////////////////////////////////
@@ -143,6 +141,8 @@ ABC_NAMESPACE_HEADER_START
 
 #endif
 
+ABC_NAMESPACE_HEADER_START
+
 /**
  * Pointer difference type; replacement for ptrdiff_t.
  * This is a signed integral type that is the same size as a pointer.
